lib/wikidata.rb

```diff
@@ -19,10 +19,16 @@ def self.ids_from_pages(lang, titles)
         token_type: false,
       }
       response = client.action :query, page_args
-      redirected_from = Hash[(response.data['redirects'] || []).map { |h| [h['to'], h['from']] }]
-      response.data['pages'].select { |_k, v| v.key? 'pageprops' }.map do |_k, v|
-        [redirected_from[v['title']] || v['title'], v['pageprops']['wikibase_item']]
+      redirected_from = Hash[(response.data['redirects'] || []).map { |h| [h['from'], h['to']] }]
```

Comment:
You're no longer _using_ this as a Hash, but instead just iterating over it, so there's no real need to cast it to one.